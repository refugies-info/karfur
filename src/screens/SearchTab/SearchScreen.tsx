import * as React from "react";
import { ScrollView } from "react-native";
import { InstantSearch } from "react-instantsearch-native";
import { StackScreenProps } from "@react-navigation/stack"
import algoliasearch from "algoliasearch/lite";
import styled from "styled-components/native";
import { Icon } from "react-native-eva-icons";
import Modal from "react-native-modal";

import { RTLTouchableOpacity } from "../../components/BasicComponents";
import { BottomTabParamList } from "../../../types"
import { useTranslationWithRTL } from "../../hooks/useTranslationWithRTL";
import SearchBox from "../../components/Search/SearchBox";
import InfiniteHits from "../../components/Search/InfiniteHits";
import { HeaderAnimated } from "../../components/HeaderAnimated";
import { FixSafeAreaView } from "../../components/FixSafeAreaView";
import { LanguageChoiceModal } from "../Modals/LanguageChoiceModal";
import { theme } from "../../theme";


const FakeInput = styled(RTLTouchableOpacity)`
  height:56px;
  width 100%;
  border-radius:${theme.radius * 2}px;
  padding:${theme.margin * 2}px;
  background-color: ${theme.colors.white};
  border: 1px solid ${theme.colors.darkGrey};
  margin-top: ${theme.margin * 4}px;
  margin-horizontal: ${theme.margin * 3}px;
  justify-content: flex-start;
  align-items: center;
`;
const FakeInputText = styled.Text`
  color: ${theme.colors.grey60};
  margin-left: ${(props: { isRTL: boolean }) => (props.isRTL ? 0 : theme.margin)}px;
  margin-right: ${(props: { isRTL: boolean }) => (props.isRTL ? theme.margin : 0)}px;
`;
const SearchModal = styled(Modal)`
  margin-horizontal: 0;
  justify-content: flex-start;
  padding-top: ${theme.margin * 6}px;
`;

const searchClient = algoliasearch("L9HYT1676M", "3cb0d298b348e76675f4166741a45599");

export const SearchScreen = ({
  navigation
}:StackScreenProps<BottomTabParamList, "Search">) => {
  const { t, isRTL } = useTranslationWithRTL();
  const [searchState, setSearchState] = React.useState({query: ""});
  const [modalOpened, setModalOpened] = React.useState(false);

  // Language modal
  const [isLanguageModalVisible, setLanguageModalVisible] = React.useState(
    false
  );
  const toggleLanguageModal = () =>
    setLanguageModalVisible(!isLanguageModalVisible);


  // Header animation
  const [showSimplifiedHeader, setShowSimplifiedHeader] = React.useState(false);
  const handleScroll = (event: any) => {
    if (event.nativeEvent.contentOffset.y > 5 && !showSimplifiedHeader) {
      setShowSimplifiedHeader(true);
      return;
    }
    if (event.nativeEvent.contentOffset.y < 5 && showSimplifiedHeader) {
      setShowSimplifiedHeader(false);
      return;
    }
    return;
  };

  return (
    <ScrollView
      style={{ flex: 1 }}
      onScroll={handleScroll}
      scrollEventThrottle={5}
    >
      <HeaderAnimated
        title={t("tabBar.Search", "Rechercher")}
        showSimplifiedHeader={showSimplifiedHeader}
        onLongPressSwitchLanguage={toggleLanguageModal}
      />

      <FakeInput onPress={() => setModalOpened(true)}>
        <Icon
          name="search-outline"
          height={24}
          width={24}
          fill={theme.colors.darkGrey}
        />
        <FakeInputText isRTL={isRTL}>
          Rechercher
        </FakeInputText>
      </FakeInput>

      <SearchModal
        isVisible={modalOpened}
        onBackdropPress={() => setModalOpened(false)}
        statusBarTranslucent={true}
        backdropColor={theme.colors.greyF7}
        backdropOpacity={1}
      >
        <FixSafeAreaView>
          <InstantSearch
            searchClient={searchClient}
            indexName="staging_refugies"
            searchState={searchState}
            onSearchStateChange={setSearchState}
          >
            <SearchBox backCallback={() => setModalOpened(false)} />
            {searchState.query !== "" &&
            <InfiniteHits
              navigation={navigation}
            />
            }
          </InstantSearch>
        </FixSafeAreaView>
      </SearchModal>

      <LanguageChoiceModal
        isModalVisible={isLanguageModalVisible}
        toggleModal={toggleLanguageModal}
      />
    </ScrollView>
  );
};
