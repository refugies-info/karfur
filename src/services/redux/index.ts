export * from "./SelectedContent";
export * from "./Themes";
export * from "./User";
