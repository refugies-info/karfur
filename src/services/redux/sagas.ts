import { SagaIterator } from "redux-saga";
import { fork } from "redux-saga/effects";
import usersSaga from "./User/user.saga";
import languagesSaga from "./Languages/languages.saga";
import contentsSaga from "./Contents/contents.saga";
import selectedContentSaga from "./SelectedContent/selectedContent.saga";

export function* rootSaga(): SagaIterator {
  yield fork(usersSaga);
  yield fork(languagesSaga);
  yield fork(contentsSaga);
  yield fork(selectedContentSaga);
}
