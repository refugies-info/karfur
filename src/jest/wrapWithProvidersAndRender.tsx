import React from "react";
import { render } from "react-native-testing-library";
import { Provider } from "react-redux";
import { createStore } from "redux";
import {
  initialRootStateFactory,
  rootReducer,
  RootState,
} from "../services/redux/reducers";

interface WrapWithProvidersAndRenderParams {
  Component: React.FunctionComponent<unknown>;
  compProps?: Record<string, unknown>;
  reduxState?: Partial<RootState>;
}

/**
 * Provide all the ugly-to-set-up providers for your component to be ready to test
 * @param Component
 * @param compProps properties passed on to [getPropsWithNavigation()] to gen navigation props
 * @param reduxState defaults to initialRootState
 */
export function wrapWithProvidersAndRender({
  Component,
  compProps = {},
  reduxState = initialRootStateFactory(),
}: WrapWithProvidersAndRenderParams) {
  const store = createStore(rootReducer, reduxState);

  const componentWithReduxAndI18n = (
    <Provider store={store}>
      <Component {...compProps} />
    </Provider>
  );

  return render(componentWithReduxAndI18n);
}
