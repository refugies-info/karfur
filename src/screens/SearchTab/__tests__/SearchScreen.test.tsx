import { SearchScreen } from "../SearchScreen";
import { wrapWithProvidersAndRender } from "../../../jest/wrapWithProvidersAndRender";
import { initialRootStateFactory } from "../../../services/redux/reducers";

jest.useFakeTimers();

jest.mock("../../../hooks/useTranslationWithRTL", () => ({
  useTranslationWithRTL: jest.fn().mockReturnValue({
    isRTL: false,
    t: jest.fn().mockImplementation((_, arg2) => arg2),
  }),
}));

jest.mock("react-native-safe-area-context", () => {
  const { SafeAreaView } = jest.requireActual("react-native-safe-area-context");

  return {
    useSafeAreaInsets: () => ({ insets: { bottom: 0 } }),
    SafeAreaView,
  };
});

jest.mock("@react-navigation/native", () => {
  return {
    useRoute: () => ({ name: "LanguageChoiceScreen" }),
  };
});

describe("Search Screen", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    (useRoute as jest.Mock).mockReturnValue({
      name: "LanguageChoiceScreen"
    });
  });

  it("should render correctly", () => {
    const component = wrapWithProvidersAndRender({
      Component: SearchScreen,
      compProps: { navigation: { goBack: jest.fn() } },
      reduxState: {
        ...initialRootStateFactory(),
      },
    });
    expect(component).toMatchSnapshot();
  });
});
