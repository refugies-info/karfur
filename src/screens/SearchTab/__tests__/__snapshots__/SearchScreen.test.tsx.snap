// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Screen should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    showShadow={false}
    style={
      Array [
        Object {
          "backgroundColor": "#F6F6F6",
          "paddingBottom": 16,
          "zIndex": 4,
        },
      ]
    }
  >
    <View
      showSimplifiedHeader={false}
      style={
        Array [
          Object {
            "backgroundColor": "#F6F6F6",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 8,
            "zIndex": 4,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "paddingTop": undefined,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "height": 124,
              "justifyContent": "flex-end",
              "paddingLeft": 24,
              "position": "relative",
            }
          }
        >
          <Text
            collapsable={false}
            style={
              Object {
                "color": "#212121",
                "fontFamily": "circularBold",
                "fontSize": 25,
                "left": 24,
                "lineHeight": 32,
                "marginLeft": 0,
                "marginRight": 8,
                "paddingBottom": 24,
                "position": "absolute",
                "textAlign": "left",
              }
            }
          >
            <Text
              style={Object {}}
            >
              Rechercher
            </Text>
            <View />
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "paddingBottom": 8,
            "paddingTop": undefined,
          }
        }
      >
        <View
          isRTL={false}
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 24,
                "paddingTop": 8,
              },
            ]
          }
        >
          <View />
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                },
              ]
            }
          >
            <View />
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#5E5E5E",
          "borderRadius": 12,
          "borderStyle": "solid",
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "flex-start",
          "marginHorizontal": 24,
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={24}
          bbWidth={24}
          fill="#5E5E5E"
          focusable={false}
          height={24}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={24}
        >
          <RNSVGGroup
            fill={4284374622}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGGroup>
              <RNSVGGroup>
                <RNSVGRect
                  height="24"
                  opacity={0}
                  width="24"
                  x={0}
                  y={0}
                />
                <RNSVGPath
                  d="M20.71 19.29l-3.4-3.39A7.92 7.92 0 0 0 19 11a8 8 0 1 0-8 8 7.92 7.92 0 0 0 4.9-1.69l3.39 3.4a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42zM5 11a6 6 0 1 1 6 6 6 6 0 0 1-6-6z"
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        isRTL={false}
        style={
          Array [
            Object {
              "color": "#5E5E5E",
              "marginLeft": 8,
              "marginRight": 0,
            },
          ]
        }
      >
        Rechercher
      </Text>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 40,
      }
    }
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollEndDrag={[Function]}
    scrollEventThrottle={20}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "marginHorizontal": 24,
          }
        }
      >
        <Text
          isRTL={false}
          style={
            Array [
              Object {
                "color": "#212121",
                "fontFamily": "circularBold",
                "fontSize": 19,
                "lineHeight": 24,
                "marginBottom": 24,
                "marginTop": 56,
                "textAlign": "left",
              },
            ]
          }
        >
          <Text
            style={Object {}}
          >
            Les fiches les plus recherchées
          </Text>
          <View />
        </Text>
      </View>
      <View>
        <Text
          isRTL={false}
          style={
            Array [
              Object {
                "color": "#212121",
                "fontFamily": "circularBold",
                "fontSize": 19,
                "lineHeight": 24,
                "marginBottom": 24,
                "marginTop": 56,
                "textAlign": "left",
              },
              Object {
                "marginHorizontal": 24,
              },
            ]
          }
        >
          <Text
            style={Object {}}
          >
            Les thèmes
          </Text>
          <View />
        </Text>
        <RCTScrollView
          contentContainerStyle={
            Object {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "marginBottom": 8,
              "paddingBottom": 8,
              "paddingHorizontal": 24,
              "width": 1100,
            }
          }
          horizontal={true}
        >
          <View>
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#443023",
                  "borderRadius": 12,
                  "display": "flex",
                  "elevation": 6,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginRight": 16,
                  "marginVertical": 8,
                  "opacity": 1,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "shadowColor": "#212121",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.16,
                  "shadowRadius": 8,
                }
              }
            >
              <Text
                isRTL={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontFamily": "circularBold",
                      "fontSize": 19,
                      "lineHeight": 24,
                      "marginLeft": 0,
                      "marginRight": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                <Text
                  style={Object {}}
                >
                  Gérer mes papiers
                </Text>
                <View />
              </Text>
              <react-native-svg
                height={20}
                style={
                  Object {
                    "stroke": "white",
                    "strokeWidth": 2,
                  }
                }
                width={20}
              />
            </View>
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#3D2884",
                  "borderRadius": 12,
                  "display": "flex",
                  "elevation": 6,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginRight": 16,
                  "marginVertical": 8,
                  "opacity": 1,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "shadowColor": "#212121",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.16,
                  "shadowRadius": 8,
                }
              }
            >
              <Text
                isRTL={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontFamily": "circularBold",
                      "fontSize": 19,
                      "lineHeight": 24,
                      "marginLeft": 0,
                      "marginRight": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                <Text
                  style={Object {}}
                >
                  Apprendre le français
                </Text>
                <View />
              </Text>
              <react-native-svg
                height={20}
                style={
                  Object {
                    "stroke": "white",
                    "strokeWidth": 2,
                  }
                }
                width={20}
              />
            </View>
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#095411",
                  "borderRadius": 12,
                  "display": "flex",
                  "elevation": 6,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginRight": 16,
                  "marginVertical": 8,
                  "opacity": 1,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "shadowColor": "#212121",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.16,
                  "shadowRadius": 8,
                }
              }
            >
              <Text
                isRTL={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontFamily": "circularBold",
                      "fontSize": 19,
                      "lineHeight": 24,
                      "marginLeft": 0,
                      "marginRight": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                <Text
                  style={Object {}}
                >
                  Apprendre un métier
                </Text>
                <View />
              </Text>
              <react-native-svg
                height={20}
                style={
                  Object {
                    "stroke": "white",
                    "strokeWidth": 2,
                  }
                }
                width={20}
              />
            </View>
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#055E5A",
                  "borderRadius": 12,
                  "display": "flex",
                  "elevation": 6,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginRight": 16,
                  "marginVertical": 8,
                  "opacity": 1,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "shadowColor": "#212121",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.16,
                  "shadowRadius": 8,
                }
              }
            >
              <Text
                isRTL={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontFamily": "circularBold",
                      "fontSize": 19,
                      "lineHeight": 24,
                      "marginLeft": 0,
                      "marginRight": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                <Text
                  style={Object {}}
                >
                  Trouver un travail
                </Text>
                <View />
              </Text>
              <react-native-svg
                height={20}
                style={
                  Object {
                    "stroke": "white",
                    "strokeWidth": 2,
                  }
                }
                width={20}
              />
            </View>
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#06508C",
                  "borderRadius": 12,
                  "display": "flex",
                  "elevation": 6,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginRight": 16,
                  "marginVertical": 8,
                  "opacity": 1,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "shadowColor": "#212121",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.16,
                  "shadowRadius": 8,
                }
              }
            >
              <Text
                isRTL={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontFamily": "circularBold",
                      "fontSize": 19,
                      "lineHeight": 24,
                      "marginLeft": 0,
                      "marginRight": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                <Text
                  style={Object {}}
                >
                  Me loger
                </Text>
                <View />
              </Text>
              <react-native-svg
                height={20}
                style={
                  Object {
                    "stroke": "white",
                    "strokeWidth": 2,
                  }
                }
                width={20}
              />
            </View>
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#910940",
                  "borderRadius": 12,
                  "display": "flex",
                  "elevation": 6,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginRight": 16,
                  "marginVertical": 8,
                  "opacity": 1,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "shadowColor": "#212121",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.16,
                  "shadowRadius": 8,
                }
              }
            >
              <Text
                isRTL={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontFamily": "circularBold",
                      "fontSize": 19,
                      "lineHeight": 24,
                      "marginLeft": 0,
                      "marginRight": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                <Text
                  style={Object {}}
                >
                  Me soigner
                </Text>
                <View />
              </Text>
              <react-native-svg
                height={20}
                style={
                  Object {
                    "stroke": "white",
                    "strokeWidth": 2,
                  }
                }
                width={20}
              />
            </View>
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#CD5605",
                  "borderRadius": 12,
                  "display": "flex",
                  "elevation": 6,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginRight": 16,
                  "marginVertical": 8,
                  "opacity": 1,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "shadowColor": "#212121",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.16,
                  "shadowRadius": 8,
                }
              }
            >
              <Text
                isRTL={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontFamily": "circularBold",
                      "fontSize": 19,
                      "lineHeight": 24,
                      "marginLeft": 0,
                      "marginRight": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                <Text
                  style={Object {}}
                >
                  Me déplacer
                </Text>
                <View />
              </Text>
              <react-native-svg
                height={20}
                style={
                  Object {
                    "stroke": "white",
                    "strokeWidth": 2,
                  }
                }
                width={20}
              />
            </View>
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#770649",
                  "borderRadius": 12,
                  "display": "flex",
                  "elevation": 6,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginRight": 16,
                  "marginVertical": 8,
                  "opacity": 1,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "shadowColor": "#212121",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.16,
                  "shadowRadius": 8,
                }
              }
            >
              <Text
                isRTL={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontFamily": "circularBold",
                      "fontSize": 19,
                      "lineHeight": 24,
                      "marginLeft": 0,
                      "marginRight": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                <Text
                  style={Object {}}
                >
                  Rencontrer des gens
                </Text>
                <View />
              </Text>
              <react-native-svg
                height={20}
                style={
                  Object {
                    "stroke": "white",
                    "strokeWidth": 2,
                  }
                }
                width={20}
              />
            </View>
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#204F01",
                  "borderRadius": 12,
                  "display": "flex",
                  "elevation": 6,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginRight": 16,
                  "marginVertical": 8,
                  "opacity": 1,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "shadowColor": "#212121",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.16,
                  "shadowRadius": 8,
                }
              }
            >
              <Text
                isRTL={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontFamily": "circularBold",
                      "fontSize": 19,
                      "lineHeight": 24,
                      "marginLeft": 0,
                      "marginRight": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                <Text
                  style={Object {}}
                >
                  Faire des études
                </Text>
                <View />
              </Text>
              <react-native-svg
                height={20}
                style={
                  Object {
                    "stroke": "white",
                    "strokeWidth": 2,
                  }
                }
                width={20}
              />
            </View>
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#073099",
                  "borderRadius": 12,
                  "display": "flex",
                  "elevation": 6,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginRight": 16,
                  "marginVertical": 8,
                  "opacity": 1,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "shadowColor": "#212121",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.16,
                  "shadowRadius": 8,
                }
              }
            >
              <Text
                isRTL={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontFamily": "circularBold",
                      "fontSize": 19,
                      "lineHeight": 24,
                      "marginLeft": 0,
                      "marginRight": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                <Text
                  style={Object {}}
                >
                  Aider une association
                </Text>
                <View />
              </Text>
              <react-native-svg
                height={20}
                style={
                  Object {
                    "stroke": "white",
                    "strokeWidth": 2,
                  }
                }
                width={20}
              />
            </View>
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#600566",
                  "borderRadius": 12,
                  "display": "flex",
                  "elevation": 6,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginRight": 16,
                  "marginVertical": 8,
                  "opacity": 1,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "shadowColor": "#212121",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.16,
                  "shadowRadius": 8,
                }
              }
            >
              <Text
                isRTL={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontFamily": "circularBold",
                      "fontSize": 19,
                      "lineHeight": 24,
                      "marginLeft": 0,
                      "marginRight": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                <Text
                  style={Object {}}
                >
                  Occuper mon temps libre
                </Text>
                <View />
              </Text>
              <react-native-svg
                height={20}
                style={
                  Object {
                    "stroke": "white",
                    "strokeWidth": 2,
                  }
                }
                width={20}
              />
            </View>
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E180A",
                  "borderRadius": 12,
                  "display": "flex",
                  "elevation": 6,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginRight": 16,
                  "marginVertical": 8,
                  "opacity": 1,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "shadowColor": "#212121",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.16,
                  "shadowRadius": 8,
                }
              }
            >
              <Text
                isRTL={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontFamily": "circularBold",
                      "fontSize": 19,
                      "lineHeight": 24,
                      "marginLeft": 0,
                      "marginRight": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                <Text
                  style={Object {}}
                >
                  Découvrir la culture
                </Text>
                <View />
              </Text>
              <react-native-svg
                height={20}
                style={
                  Object {
                    "stroke": "white",
                    "strokeWidth": 2,
                  }
                }
                width={20}
              />
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  />
</View>
`;
